pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py

```diff
@@ -439,6 +441,9 @@ def update_logger_connector(self, fx_name: str = None) -> None:
             epoch_log_metrics = self.get_epoch_log_metrics()
             logger_connector.logged_metrics.update(epoch_log_metrics)
             logger_connector.logged_metrics.update(epoch_dict)
+            if not self.trainer.running_sanity_check and not is_train:
+                if len(epoch_log_metrics) > 0:
+                    self.trainer.dev_debugger.track_logged_metrics_history(deepcopy(epoch_log_metrics))
```

Comment:
Does this end up storing all the previous results in memory?

Reply:
No, only one. So pretty light. I had to it as `track_logged_metrics_history` is in-place operation and adding `global_step` key which was breaking. And it will call only dev_debugger is activated :)